__tests__/services/targets.service.spec.ts

```diff
@@ -101,4 +102,36 @@ describe('TargetsService', () => {
         .rejects.toThrowError(TargetNotSavedException);
     });
   });
+
+  describe('listTargets', () => {
+    beforeEach(async () => {
+      user = await factory(User)().create();
+      target = await factory(Target)().create({ userId: user.id });
+    });
+
+    it('should return a list of targets for the user', async () => {
+      jest.spyOn(targetRepository, 'find')
+        .mockResolvedValueOnce([target]);
+
+      const response = await targetsService.listTargets(user.id);
+      expect(response).toEqual([target]);
```

Comment:
You could also test if the returned object is an instance of the Target object.

Reply:
Thank you, nice observation ! I added that type check.